DataBiosphere/Saturn-UI
# WX-1756 Fix resource monitoring parameter behavior

### Jira Ticket: [WX-1756](https://broadworkbench.atlassian.net/browse/WX-1756)

### What
- Check resource monitoring enablement before sending parameters with workflow launch request

### Why
- Previously, populating string values in the resource monitoring section would continue sending the strings even if monitoring was disabled.
- We made users think the show/hide checkbox was an on/off toggle; now it really is an on/off toggle

Before:

```
{
  "deleteIntermediateOutputFiles": false,
  "entityName": "some_1",
  "entityType": "participant",
  "ignoreEmptyOutputs": false,
  "methodConfigurationName": "localization-test",
  "methodConfigurationNamespace": "oednichols",
  "monitoringImage": "asdf",
  "monitoringImageScript": "asdf",
  "monitoringScript": "asdf",
  "useCallCache": true,
  "useReferenceDisks": false,
  "userComment": ""
}
```

After:

```
{
  "deleteIntermediateOutputFiles": false,
  "entityName": "some_1",
  "entityType": "participant",
  "ignoreEmptyOutputs": false,
  "methodConfigurationName": "localization-test",
  "methodConfigurationNamespace": "oednichols",
  "useCallCache": true,
  "useReferenceDisks": false,
  "userComment": ""
}
```

### Testing strategy

- Tested by trying to shake out sequence of events bugs; manually launching workflows and verifying terminal status.

1. Nominal with no monitoring selected - no monitoring parameters sent
2. Enable monitoring - params sent as expected
3. Disable monitoring - no monitoring parameters sent
4. Re-enable monitoring - params sent as expected


[WX-1756]: https://broadworkbench.atlassian.net/browse/WX-1756?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
